### ISSUE TITLE
Air conditioning in Homekit

### ISSUE THREAD

**Issue by johngrinders** [swttt/com.swttt.homekit#175]
Good afternoon, literally today I installed the application from apps.athom.com Melcloud. Everything works fine through Homey. I decided to add this air conditioner through Homeykit to Homekit. Everything went well, everything was added as a new device in which there is a choice of temperature, a choice of mode, but unfortunately all this does not work for some reason. I wanted to clarify this problem on the side of the Melcloud developer or all the same, the problem may be in your plugin. Thank!!!

**Comment by robertklep** [comment]
Can you clarify what exactly doesn't work? Does the device show up in the Home app on your phone, but you cannot control it?

**Comment by johngrinders** [comment]
The device itself is in Homekit, but it does not update statuses and I can’t control the air conditioner through it.

**Comment by robertklep** [comment]
Can you find the device [on this page](https://developer.athom.com/tools/devices) and tell me its "Class" and the capability names?

**Comment by johngrinders** [comment]
Class - thermostat
Capabilities:
onoff | boolean | false | truefalse | 13 minutes ago
mode_capability | enum | fan | offheatdrycoolfanauto | 13 minutes ago
target_temperature | number | 19 |   | 13 minutes ago
measure_temperature | number | 25 |   | 13 minutes ago
fan_power | number | 3 |   | 13 minutes ago
vertical | enum | auto | autotopmiddletopmiddlemiddlebottombottomswing | 13 minutes ago
horizontal | enum

**Comment by robertklep** [comment]
For devices of class `thermostat`, HomeyKit supports `measure_temperature` and `target_temperature` only. Those also don't work?

**Comment by johngrinders** [comment]
Yes, only temperature works, the rest: turning on, turning off, changing modes unfortunately does not work.

**Comment by robertklep** [comment]
Those capabilities are not supported because the HomeKit protocol doesn't support them for thermostat devices.

**Comment by johngrinders** [comment]
OK. Thank!!!

**Comment by XattSPT** [comment]
Thanks @robertklep. I've modified the  Melcloud app to assure homeykit compatibility using thermosat_mode capability. The bug was in 'my side'

**Comment by robertklep** [comment]
@XattSPT thanks!
